microsoft/testsuites/sysbench/sysbenchperf.py

```diff
@@ -0,0 +1,406 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import json
+import re
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, Dict, List
+
+from assertpy.assertpy import fail
+
+from lisa import (
+    Environment,
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    notifier,
+)
+from lisa.messages import TestStatus, send_sub_test_result_message
+from lisa.testsuite import TestResult
+from lisa.tools import Sysbench
+from lisa.util.process import ExecutableResult
+
+
+@dataclass
+class SysbenchTestResult:
+    name: str = ""
+    status: TestStatus = TestStatus.QUEUED
+    message: str = ""
+
+
+@TestSuiteMetadata(
+    area="sysbench",
+    category="performance",
+    description="""
+    This test suite is for executing the sysbench tests
+    """,
+)
+class SysbenchPerfTestSuite(TestSuite):
+    @TestCaseMetadata(
+        description="""
+            Runs Sysbench test for cpu
+        """,
+        priority=3,
+    )
+    def verify_sysbench_cpu_perf(
+        self,
+        log: Logger,
+        node: Node,
+        environment: Environment,
+        log_path: Path,
+        result: TestResult,
+        variables: Dict[str, Any],
+    ) -> None:
+        subtest_name: str = "default_cpu_perf_check"
+        res: SysbenchTestResult = SysbenchTestResult()
+
+        output: ExecutableResult = node.tools[Sysbench].run_cpu_perf()
+
+        if output.exit_code != 0:
+            res.name = subtest_name
+            res.message = f"{subtest_name} failed with " f"exit code {output.exit_code}"
+            res.status = TestStatus.FAILED
+        else:
+            res = process_cpu_perf_result(output.stdout, subtest_name)
```

Comment:
If the pattern is always used like "run and process". It can be in `run_cpu_perf`, so the caller is simple.

Reply:
Yes pattern is run the command and process the stdout to extract the result according to testcase.

I created wrote result-extraction in such a way that we can re-use it for common parameters. All `process_*_perf_result` would call `process_perf_result` to get common result metrics.

Also in future we might visit the testcase where we would be running sysbench with more permutation combination of parameters for cpu/memory/fileio. This way, it would require minimal changes.

Reply:
The `process_cpu_perf_result` can be called in `run_cpu_perf`. So all logic are handled by tool level. The subtest results should be generated whenever the tool runs. It can still add other changes in either method.

Reply:
Done. Please check the latest push.